src/java/voldemort/client/protocol/admin/AdminClient.java

```diff
@@ -1875,10 +1877,16 @@ public void verifyOrAddStore(StoreDefinition newStoreDef, String localProcessNam
                 try {
                     // Get all StoreDefinitions from each nodes in the cluster
                     remoteStoreDefs = metadataMgmtOps.getRemoteStoreDefList(nodeId).getValue();
-                } catch (UnreachableStoreException e) {
-                    logger.warn("Failed to contact " + node.briefToString() + " in order to validate the StoreDefinition.");
-                    unreachableNodes.add(node);
-                    continue;
+                } catch (Exception e) {
+                    if (ExceptionUtils.recursiveClassEquals(e,
```

Comment:
Can you leave a comment here saying that?

Admin Client Checkout code may throw, Connect, Socket or EOFException, so checking for IOException as well.